packages/devextreme/js/__internal/ui/splitter/resize_handle.ts

```diff
@@ -0,0 +1,119 @@
+import { ResizeEndEvent, ResizeStartEvent } from '@js/ui/resizable';
+
+import Guid from '../../../core/guid';
+import { extend } from '../../../core/utils/extend';
+import eventsEngine from '../../../events/core/events_engine';
+import { end as dragEventEnd, move as dragEventMove, start as dragEventStart } from '../../../events/drag';
+import { addNamespace } from '../../../events/utils/index';
+import Widget from '../../../ui/widget/ui.widget';
+
+const RESIZE_HANDLE_CLASS = 'dx-resize-handle';
+const HORIZONTAL_DIRECTION_CLASS = 'dx-resize-handle-horizontal';
+const VERTICAL_DIRECTION_CLASS = 'dx-resize-handle-vertical';
+const RESIZE_HANDLER_MODULE_NAMESPACE = 'dxResizeHandle';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+class ResizeHandle extends (Widget as any) {
+  _getDefaultOptions(): Record<string, unknown> {
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+    return extend(super._getDefaultOptions(), {
+      direction: 'horizontal',
+      onResize: null,
+      onResizeEnd: null,
+      onResizeStart: null,
+    });
+  }
+
+  _init(): void {
+    super._init();
+    const namespace = `${RESIZE_HANDLER_MODULE_NAMESPACE}${new Guid().toString()}`;
+    this.RESIZE_START_EVENT_NAME = addNamespace(dragEventStart, namespace);
+    this.RESIZE_EVENT_NAME = addNamespace(dragEventMove, namespace);
+    this.RESIZE_END_EVENT_NAME = addNamespace(dragEventEnd, namespace);
+    this.handlers = {};
+  }
+
+  _initMarkup(): void {
+    super._initMarkup();
+
+    this._toggleDirection();
+    this.$element().addClass(RESIZE_HANDLE_CLASS);
+  }
+
+  _toggleDirection(): void {
+    this.$element().toggleClass(HORIZONTAL_DIRECTION_CLASS, this._isHorizontalDirection());
+    this.$element().toggleClass(VERTICAL_DIRECTION_CLASS, !this._isHorizontalDirection());
+  }
+
+  _render(): void {
+    super._render();
+
+    this._attachEventHandlers();
+    this._renderActions();
+  }
+
+  _resizeStartHandler(e: ResizeStartEvent): void {
+    this._resizeStartAction({
+      event: e,
+    });
+  }
+
+  // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+  _resizeHandler(e): void {
+    this._resizeAction({
+      event: e,
+    });
+  }
+
+  _resizeEndHandler(e: ResizeEndEvent): void {
+    this._resizeEndAction({
+      event: e,
+    });
+  }
+
+  _renderActions(): void {
+    this._resizeStartAction = this._createActionByOption('onResizeStart');
```

Comment:
Should we add exclude validators?
```
{
       excludeValidators: ['readOnly'] // disabled
 }
```
You may give answer later. Please extract the separate subtask.